### PR TITLE
Replace Travis CI with GitHub Actions

### PR DESCRIPTION
**Pull request by rojtjo**

### CODE REVIEW

**Comment by HVSoftware** [comment]
Is there a reason why we cannot use this MR? It will be nice if we can improve this script with phpcs or phpstan checks. At least this will keep the code clean.

**Comment by rojtjo** [comment]
The only thing currently missing is testing with multiple versions of dependencies such as `guzzlehttp/guzzle` and `ramsey/uuid`. We might want to include this before merging.

Let me know if this is needed and I'll attempt to set it up.

**Comment by HVSoftware** [comment]
Do you want to test all the versions of these two packages or is testing just the lowest and latest packages enough?

I saw you used the composer update within the script. This will use the latest version to the `guzzlehttp/guzzle` and `ramsey/uuid` packages. 

With the following command you are able to test the lowest versions. 

- composer update --prefer-lowest

Maybe we can just add the following steps:

- name: Install lowest dependencies
  - composer update --prefer-lowest 
- name: Execute tests with lowest packages
- name: Analyse files with lowest packages
- name: Install latest dependencies
  - composer update
- name: Execute tests with latest packages
- name: Analyse files with latest packages

This will atleast covers the boundaries.

**Comment by rojtjo** [comment]
I've added stability flags to the matrix to test lowest and stable.

In order to get the tests to green I had to remove the unused `dg/bypass-finals` dependency and bump `guzzlehttp/guzzle` to `^6.4`.

Next up would be to pull in the changes from master and run psalm as well.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                  echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+            - name: Validate composer.json
+              run: composer validate --strict
+
+            - name: Lint files
+              run: find src tests -name '*.php' | xargs -n 1 -P4 php -l
+
+            - name: Install dependencies
+              run: composer update --${{ matrix.stability }} --prefer-dist --no-interaction
```
Comment (HVSoftware) [inline .github/workflows/tests.yml:39]: Nice solution and always nice to remove the if checks. 👍🏻

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+            - name: Execute tests
+              run: vendor/bin/phpunit --fail-on-risky
```
Comment (HVSoftware) [inline .github/workflows/tests.yml:42]: Just a question. Will phpunit be ran for all versions of php with the preferred stability?

Comment (rojtjo) [reply .github/workflows/tests.yml:42]: Yes, it uses the dependencies that were installed in the previous step

**Comment by manuelderuiter** [comment]
Looks great! Thank you!

**Pull request merged**